tests/integration/test_records.py

```diff
@@ -425,3 +425,25 @@ def test_responses(client, simple_ensemble):
     assert len(responses) == 1
     data = client.get(f"/records/{responses['rec3']['id']}/data").json()
     assert data == [[1, 2], [4, 5]]
+
+
+def test_fetch_matrix_ensemble_record_by_realization(client, simple_ensemble):
+    ensemble_id = simple_ensemble(parameters=[], responses=["polynomial_output"])
+    client.post(
+        url=f"/ensembles/{ensemble_id}/records/polynomial_output/matrix",
+        headers={"content-type": "text/csv"},
+        data=b",0,1\n0,1.0,3.0\n1,1.0,1.0\n2,2.0,4.0\n3,3.0,1.0\n4,5.0,5.0\n",
+    )
+    resp = client.get(
+        url=f"/ensembles/{ensemble_id}/records/polynomial_output",
+        params={"realization_index": 0},
```

Comment:
It might work, but I do consider this ambiguous though. It's nice to reuse the variable (`realization_index`) but I would rather use a standalone parameter (eg. `row=0`) just in case that `polynomial_output` was stored for each realization separately.

Reply:
How come this is ambiguous? Is not the idea that the rows of the matrix corresponds to the different realisations?

Reply:
Yes, that works when we have some definition of record matrix, ie., what you wrote above. My concern was rather if we have for instance a realization record matrix and thus having a more general API to account for such cases.